BlockEQ/View Controllers/Wallet/AddressEntryViewController.swift

```diff
@@ -70,8 +70,26 @@ final class AddressEntryViewController: UIViewController {
         let availableSendBalance = account.availableSendBalance(for: asset).displayFormattedString
         navigationItem.title = String(format: "TRADE_BALANCE_FORMAT".localized(), availableSendBalance, asset.shortCode)
     }
+    
+    @IBAction func textFieldEditingDidChange(_ sender: Any) {
+        let text = sendAddressTextField.text!
+        if (text.contains("*") && text.contains(".com")) {
+            Federation.resolve(stellarAddress: text) { (response) -> (Void) in
```

Comment:
I try to keep this type of logic at the coordinator level so that my view controllers can stay dumb and can be updated at any time with any data. That way, I don't need to worry about asynchronous code in my view updating.

It requires a lot more delegation back to the parent coordinator, but is worth the testability.

Reply:
You might do something like: `delegate?.requestedAddressLookup(self, address)`, and then have a delegate implementation method that updates the appropriate text field with the account ID.